### PR TITLE
Add top level, sram and assembly modules

### PR DESCRIPTION
**Pull request by meetshah1995**
Added the core, with the mainframe tests ongoing, will keep adding the tests  to this PR.

### CODE REVIEW

File: riscv/core.py [hunk 1]
```diff
@@ -64,12 +64,13 @@ def vscale_core(clock,
...
     dmem_badmem_e = Signal(intbv(0)[1:])
 
-    @always_comb
+    @instance
     def assign():
         htif_ipi_req_valid.next = intbv(0)[1:]
         htif_ipi_req_data.next = intbv(0)[1:]
         htif_ipi_resp_ready.next = intbv(1)[1:]
         htif_debug_stats_pcr.next = intbv(0)[1:]
+        yield delay(1)
```
Comment (jck) [inline riscv/core.py]: delay isn't synthesizable.

File: riscv/top_level.py [hunk 2]
```diff
@@ -62,7 +57,11 @@ def top_level(clk,
     resetn = not reset
     htif_reset = reset
 
-    vscale_core = core(clk=clk,
+    @always(delay(10))
```
Comment (jck) [inline riscv/top_level.py]: The clock isn't part of the core. Drive the clock from the testbench.

File: tests/test_mult_div.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# from myhdl import block, Signal, modbv, instance, always, delay
+# from riscv.md_constants import *
+# from riscv.mult_div import mult_div
+# from riscv.opcode_constants import XPR_LEN
+#
+#
+# @block
+# def test_mult_div():
+#     clock, reset, req_valid, req_ready, req_in_1_signed, req_in_2_signed = [Signal(False) for _ in range(6)]
+#     req_op = Signal(modbv(0)[MD_OP_WIDTH:])
```
Comment (jck) [inline tests/test_mult_div.py]: Why is this test commented? If the test is incomplete or doesn't pass yet, you can use py.test's xfail

Comment (meetshah1995) [inline tests/test_mult_div.py]: Yes will use xfail. @srivatsan-ramesh

File: riscv/top_level.py (hunk 2 above)
Comment (meetshah1995) [inline riscv/top_level.py]: Yes will isolate the clock from the module.

File: riscv/core.py (hunk 1 above)
Comment (meetshah1995) [inline riscv/core.py]: Actually from the logic, there is no need for it, just added it as a safe-side measure. Will remove it.

**Pull request merged**